nixos/lib/test-driver/test-driver.py

```diff
@@ -597,7 +597,7 @@ def create_socket(path):
 
         def process_serial_output():
             for line in self.process.stdout:
-                line = line.decode().replace("\r", "").rstrip()
+                line = line.decode(errors="ignore").replace("\r", "").rstrip()
```

Comment:
is this just to safe guard against binary output?

Reply:
In the boot tests there comes colored output from the boot menu selection. The decoder didn't like it and dropping such characters solved the problem.

Reply:
It's probably escape sequences in general, not just color codes.

We should try to `decode('unicode_escape')` instead of throwing them away - it might be useful to keep them, at least if we want to render colored text in the html output at some point…

Reply:
Ok the boot menu messages are colored but the colors were not the actual problem. you see colors in the terminal when you run the boot tests with the current patch set.

But let me see what i can do with `unicode_escape`, i did not know about it.

Reply:
ok looks good. There are more ugly characters that make no sense on the serial then, but it's of course of no harm.